src/utils/SetupUtil.js

```diff
@@ -82,6 +82,9 @@ export default {
           machineVersion
         });
 
+        virtualBox.getShareDir(machine.name());
```

Comment:
What you are comment is totally right. I think that the operation of add a new folder to the machine is not a "normal" activity when you are working, it is more usual an activity you do so few times when you are configuring your enviroment. So, I think recovering once when the program starts is enough, but if you think it's better another approach we can do this call when anyone is trying to change the volume.. 

I'm open to all your considerations, you are the master here :)